<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogBasicSettings</class>
 <widget class="QDialog" name="DialogBasicSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Basic Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="8" column="3">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="2" column="3">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_1">
      <attribute name="title">
       <string>Common</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="hlayout_l1">
         <item>
          <widget class="QGroupBox" name="horizontalGroupBox_3">
           <layout class="QHBoxLayout" name="horizontalLayout_19">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Listen Address</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="inbound_address"/>
            </item>
            <item>
             <widget class="QPushButton" name="inbound_auth">
              <property name="text">
               <string notr="true"/>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupbox_custom_inbound">
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Custom Inbound</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="custom_inbound_edit">
              <property name="text">
               <string>Edit</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="hlayout_l2">
         <item>
          <widget class="QGroupBox" name="horizontalGroupBox_2">
           <layout class="QHBoxLayout" name="horizontalLayout_13">
            <item>
             <widget class="QLabel" name="inbound_socks_port_l">
              <property name="text">
               <string>Socks Listen Port</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="inbound_socks_port">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_http">
           <layout class="QHBoxLayout" name="horizontalLayout_16">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>HTTP Listen Port</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="inbound_http_port">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="http_enable">
              <property name="text">
               <string>Enable</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="hlayout_l4">
         <item>
          <widget class="QGroupBox" name="groupBox_log">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string notr="true">Loglevel</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="log_level"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox1">
         <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="0,8,1,1">
          <item>
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Test URL</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="test_url"/>
          </item>
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Concurrent</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="test_concurrent"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="horizontalGroupBox1">
         <layout class="QHBoxLayout" name="horizontalLayout_18">
          <item>
           <widget class="QCheckBox" name="check_include_pre">
            <property name="text">
             <string>Include Pre-release when checking update</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="sys_proxy_format_vline">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="sys_proxy_format">
            <property name="text">
             <string>System proxy format</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="horizontalWidget_1" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Style</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="style_h_1">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_8">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Theme</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="theme">
              <item>
               <property name="text">
                <string>System</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">flatgray</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">lightblue</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">blacksoft</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="set_custom_icon">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Set custom icon</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox2">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_15">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string notr="true">Language</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="language">
              <item>
               <property name="text">
                <string notr="true">System</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">English</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">简体中文</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">فارسی</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <layout class="QHBoxLayout" name="style_h_2">
         <item>
          <widget class="QGroupBox" name="traffic_statistics_box">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Statistics refresh rate</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="rfsh_r">
              <item>
               <property name="text">
                <string>Fast</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Slow</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Off</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="connection_statistics_box">
           <layout class="QHBoxLayout" name="horizontalLayout_15">
            <item>
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Connection statistics</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="connection_statistics">
              <property name="text">
               <string>Enable</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <layout class="QHBoxLayout" name="style_h_3">
         <item>
          <widget class="QGroupBox" name="horizontalGroupBox_5">
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QCheckBox" name="start_minimal">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Hide dashboard at startup</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <layout class="QHBoxLayout" name="horizontalLayout_23">
            <item>
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>Max log lines</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="max_log_line">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Subscription</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>User Agent</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="MyLineEdit" name="user_agent"/>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="sub_use_proxy">
         <property name="text">
          <string>Use proxy when updating subscription</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QCheckBox" name="sub_insecure">
         <property name="text">
          <string>Ignore TLS errors when updating subscription</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QCheckBox" name="sub_clear">
         <property name="text">
          <string>Clear servers before updating subscription</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Core</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox_core">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string notr="true">core_name</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_22">
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Asset Location</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="MyLineEdit" name="core_v2ray_asset">
              <property name="placeholderText">
               <string>Default: dir of &quot;nekoray&quot;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="core_v2ray_asset_pick">
              <property name="text">
               <string>Select</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_21">
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Switch core</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="switch_core_v2ray">
              <property name="text">
               <string notr="true">V2Ray</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="switch_core_sing_box">
              <property name="text">
               <string notr="true">sing-box</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_mux">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_17">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Multiplex (mux)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="mux_protocol"/>
          </item>
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>concurrency</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="mux_concurrency"/>
          </item>
          <item>
           <widget class="QCheckBox" name="mux_default_on">
            <property name="text">
             <string>Default On</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="core_settings">
         <property name="text">
          <string>Core Options</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>Extra Core</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QScrollArea" name="extra_core_box_scrollArea">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="extra_core_box_scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>198</width>
            <height>58</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QWidget" name="horizontalWidget_4" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QPushButton" name="extra_core_add">
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="extra_core_del">
                <property name="text">
                 <string>Delete</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>Security</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QCheckBox" name="skip_cert">
         <property name="text">
          <string>Skip TLS certificate authentication by default (allowInsecure)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="horizontalGroupBox3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Enable hook.js</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="enable_js_hook">
            <item>
             <property name="text">
              <string notr="true">Disable</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">Enable</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">Enable + load std module</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="horizontalGroupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_26">
          <item>
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Default uTLS Fingerprint</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="utlsFingerprint">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyLineEdit</class>
   <extends>QLineEdit</extends>
   <header>ui/widget/MyLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogBasicSettings</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogBasicSettings</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
